enterprise/cmd/repo-updater/internal/authz/perms_syncer.go

```
@@ -988,13 +1047,14 @@ func (s *PermsSyncer) syncPerms(ctx context.Context, logger log.Logger, syncGrou
 			metricsConcurrentSyncs.WithLabelValues(request.Type.String()).Inc()
 			defer metricsConcurrentSyncs.WithLabelValues(request.Type.String()).Dec()
 
-			err := runSync()
+			providerStates, err := runSync()
 			if err != nil {
```

I think we should extend this check to log a potential error in `providerStates` that could be there even if we do not have `err` returned from the sync. Otherwise we would still potentially swallow a provider error.

---

Forcibly creating an error log here would override error spam mitigation efforts elsewhere